swagger.yaml

```diff
@@ -12,29 +12,73 @@ tags:
 schemes:
 - "https"
 
+host: "api.oregonstate.edu"
+
+basePath: "/v1"
+
+externalDocs:
+  description: "GitHub Repository"
+  url: "https://github.com/osu-mist/core-permissions-api"
+
+consumes:
+  - "application/json"
+
+produces:
+  - "application/json"
+
 paths:
-  /permissions/{osuID}:
 
+  /permissions/{osuID}:
     get:
       tags:
       - "permissions"
-      summary: "Retrieve permissions for user by OSU ID"
+      summary: "Retrieve permissions for user by their OSU ID"
       operationId: "getPermissionsById"
-      produces:
-      - "application/json"
       parameters:
         - name: "osuID"
           in: "path"
           description: "OSU ID of person whose permissions will be retrieved"
           required: true
           type: "integer"
+
       responses:
         200:
           description: "Permissions successfully retrieved"
           schema:
             $ref: "#/definitions/PermissionsResult"
         404:
-          description: "No user with this OSU ID found"
+          description: "No permissions found for this OSU ID"
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: "Internal server error"
+          schema:
+            $ref: "#/definitions/Error"
+
+  /permissions:
+    get:
+      tags:
+      - "permissions"
+      summary: "Retrieve permissions for users by parameters"
+      operationId: "getPermissions"
+      parameters:
+        - name: "osuID"
+          in: "query"
+          description: "OSU ID of person whose permissions will be retrieved"
+          required: false
+          type: "integer"
+        - name: "onid"
+          in: "query"
+          description: "ONID of person whose permissions will be retrieved"
+          required: false
+          type: "string"
+      responses:
+        200:
+          description: "Permissions successfully retrieved"
+          schema:
+            $ref: "#/definitions/PermissionsResultArray"
+        400:
+          description: "Invalid syntax in parameters"
```

Comment:
When do you expect to get a 400 response? We should also send a 400 if `osuID` and `onid` are both included since it may be possible to refer to different people. Similarly, when `osuID` and `onid` are both missing, we should return a 400 as well.

Reply:
Ok, it makes sense to return a 400 in both of these cases. I was also thinking that a 400 would be returned if the query has invalid types for either of the fields.